.github/workflows/cmake.yml

```diff
@@ -0,0 +1,161 @@
+name: CMake on multiple platforms
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations. Consider changing this to true when your workflow is stable.
+      fail-fast: false
+
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.10', '3.11']
```

Comment:
I would recommend building only the native library here using a build matrix, then a separate job to build the Python binary wheels.